config/management/operational/src/validator_config.rs

```diff
@@ -156,7 +156,8 @@ impl RotateKey {
         // current key (to resynchronize the validator config on the blockchain).
         let mut storage_key = storage.ed25519_public_from_private(key_name)?;
         let keys_match = match key_name {
-            CONSENSUS_KEY => storage_key == validator_config.consensus_public_key,
+            // Rotate bls12381 is not supported
```

Comment:
As a curiosity, why is this not supported?

Reply:
too lazy to support it, similar to gas. someone needs to work on the key rotation story

Reply:
But why couldn't we use the previous Ed25519 key rotation mechanism... shouldn't it be pretty oblivious of the actual signature scheme?

Reply:
well, the specific vault implementation dictates it's a Ed25519 key https://github.com/aptos-labs/aptos-core/blob/a51f98f5e0752f51f753de8bae68163e5316566b/secure/storage/src/vault.rs#L271

changing the interface will mess it up, I'd rather someone deleted vault and support rotation properly

Reply:
I understand. It feels important to support key rotation for validators. I hope we add it back.